### DISCUSSION TITLE
Anyone knows data directory of `perm` in table `t_ds_relation_udfs_user` ?

### DISCUSSION THREAD

**Discussion by chengqiZ** [apache/dolphinscheduler#10423] [Q&A] [answered]
i wonder what means of the column `perm` in `t_ds_relation_udfs_user` . 
Can someone tell me What does 0-7 of perm field in the table `t_ds_relation_udfs_user` mean respectively？

**Comment by zhongjiajie** [comment, marked as answer]
This is an internal variable, usually only dolphinscheduler understands it is enough， but you could still find then in https://github.com/apache/dolphinscheduler/blob/a72742b9de0bec62fa5bf0e3cd8553704faacac9/dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/entity/UDFUser.java#L90-L92 , and the most usage is two of them https://github.com/apache/dolphinscheduler/blob/f6fea06f1051f5962106e6299eea9696bddf12d4/dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/Constants.java#L737-L741

**Reply by chengqiZ** [reply]
Okay, got it
